solution/build.sh

```diff
@@ -97,6 +97,72 @@ function copy_image()
         dir:${FULL_PATH}
 }
 
+function generate_local_dashboard()
+{
+    if [[ $# -ne 1 ]]
+    then
+        echo "missing argument, $0 <dashboard>"
+        exit 1
+    fi
+
+    file=$1
+    filename=$(basename ${file})
+
+    dashboard_base_dir=${IMAGES_ROOT}/${filename%.json}-dashboard/${VERSION}/
+    mkdir -p ${dashboard_base_dir}
+
+    # get sha256, sha256sum prints the checksum and the filename, keep the checksum only
+    digest=$(sha256sum  ${file} | cut -d " " -f 1)
+    # get only the size (in bytes) of the file. format "%s" only prints the size
+    size=$(stat --printf "%s" ${file})
+
+    # generate empty config file
+    config_file=${dashboard_base_dir}/config.json
+    echo "{}" > ${config_file}
+    config_sum=$(sha256sum  ${config_file} | cut -d " " -f 1)
+    mv ${config_file} ${dashboard_base_dir}/${config_sum}
+
+    cp ${file} ${dashboard_base_dir}/${digest}
+
+    cat > ${dashboard_base_dir}/manifest.json <<EOF
+{
+    "schemaVersion": 2,
+    "mediaType": "application/vnd.oci.image.manifest.v1+json",
+    "config": {
+        "mediaType": "application/vnd.oci.image.config.v1+json",
+        "digest": "sha256:${config_sum}",
+        "size": 2
+    },
+    "layers": [
+        {
+            "mediaType": "application/grafana-dashboard+json",
+            "digest": "sha256:${digest}",
+            "size": ${size},
```

Comment:
i would suggest introducing a function which generates this output from an input stream, computes the sha256sum, renames the file and generates this block:

```
# $1 : mime-type of the file
# $2 : optionally, name of the file
# input : the content of the file
# output : the manifest fragment defining the file
function processFile {
   local tmp=$(mktemp xxxxx)
   cat > $tmp
    digest=$(sha256sum  ${tmp} | cut -d " " -f 1)     # get sha256, sha256sum prints the checksum and the filename, keep the checksum only
    size=$(stat --printf "%s" ${tmp})     # get only the size (in bytes) of the file. format "%s" only prints the size
    mv $tmp $digest
    
    echo "mediaType": "$1",
    echo "digest": "sha256:${digest}",
    echo "size": ${size},
    [ "$2" ] && echo "\"annotations\": { \"org.opencontainers.image.title\": \"$2\" }"
}
```

It can then be used both the the config and the dashboard:

```
$(processFile "application/vnd.oci.image.config.v1+json" <<< '{}')
$(processFile "application/grafana-dashboard+json" "$filename" < "$filename")

Reply:
I thought I'll do this later, for now we don't have any dashboards for others than third parties so once we are clear on the number of dashboards, if we have multiple dashboards, if we need to etc we can re-work the function that produces the manifest file. for now we don't know exactly what we are gonna need.

Reply:
done